#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_Makepkg_DESTDIR.dpatch by  <thibaut@tidirium.alliance.ak>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make Makepkg DESTDIR-aware --- Yorick comes with a nice system for 
## DP: easily building plugin packages. However, this system always installs 
## DP: the build package into the system Y_SITE/Y_HOME. This patch makes it 
## DP: possible to give a destination root to make (DESTDIR variable) and even 
## DP: to specify the target Y_SITE and Y_HOME directories (DEST_Y_SITE, 
## DP: DEST_Y_HOME and DEST_Y_BINDIR). The former is very useful for building 
## DP: Debian packages of a plugin (make DESTDIR=debian/<pkg-name> install), 
## DP: while the later is useful for admins who wish to install home-build 
## DP: plugins under /usr/local or users who wish to install in thir home 
## DP: directory (e.g. make DEST_Y_SITE=/usr/local/lib/yorick/ 
## DP: DEST_Y_HOME=/usr/local/lib/yorick/ install).

@DPATCH@

diff -urN orig/Makepkg tp/Makepkg
--- orig/Makepkg	2006-05-05 13:31:27.673761552 +0200
+++ tp/Makepkg	2006-05-05 10:02:34.000000000 +0200
@@ -39,6 +39,7 @@
 # CODGER, PKG_NAME, PKG_I, PKG_I_DIR, EXTRA_PKGS
 # Y_EXE, Y_SITE, Y_HOME
 # PKG_CLEAN, Y_BINDIR
+# DESTDIR, DEST_Y_SITE, DEST_Y_HOME, DEST_Y_BINDIR
 
 # defined in package Makefile (which includes this Makepkg):
 # Y_MAKEDIR  directory containing Make.cfg, Makepkg, Makeexe, Makedll
@@ -66,6 +67,11 @@
 # LIB_TARGETS  optional prerequisites for static library
 # DLL_TARGETS  optional prerequisites for dynamic library
 
+# DESTDIR       optional (un)install root
+# DEST_Y_SITE   optional install Y_SITE (default: $(DESTDIR)$(Y_SITE))
+# DEST_Y_HOME   optional install Y_HOME (default: $(DESTDIR)$(Y_HOME))
+# DEST_Y_BINDIR optional install Y_BINDIR (default: $(DESTDIR)$(Y_BINDIR))
+
 # ------------------------------------------------------------------------
 
 # defined in Make.cfg:
@@ -151,23 +157,27 @@
 
 Y_GROUP=`cat $(Y_LIBEXE)/install.grp`
 YNSTALL=$(Y_LIBEXE)/install.sh $(Y_GROUP)
+DESTDIR=
+DEST_Y_SITE=$(DESTDIR)$(Y_SITE)
+DEST_Y_HOME=$(DESTDIR)$(Y_HOME)
+DEST_Y_BINDIR=$(DESTDIR)$(Y_BINDIR)
 install:: install-$(TGT)
-	$(YNSTALL) $(PKG_I_DEPS) $(Y_SITE)/i0
-	if test -n "$(PKG_I_START)"; then $(YNSTALL) $(PKG_I_START) $(Y_HOME)/i-start; fi
-	if test -n "$(PKG_I_EXTRA)"; then $(YNSTALL) $(PKG_I_EXTRA) $(Y_SITE)/i; fi
+	$(YNSTALL) $(PKG_I_DEPS) $(DEST_Y_SITE)/i0
+	if test -n "$(PKG_I_START)"; then $(YNSTALL) $(PKG_I_START) $(DEST_Y_HOME)/i-start; fi
+	if test -n "$(PKG_I_EXTRA)"; then $(YNSTALL) $(PKG_I_EXTRA) $(DEST_Y_SITE)/i; fi
 install-exe: $(PKG_EXE)
-	$(YNSTALL) $(PKG_EXE) $(Y_BINDIR)
-	$(YNSTALL) $(PKG_LIB) $(Y_HOME)/lib
-	$(RANLIB) $(Y_HOME)/lib/$(PKG_LIB)
+	$(YNSTALL) $(PKG_EXE) $(DEST_Y_BINDIR)
+	$(YNSTALL) $(PKG_LIB) $(DEST_Y_HOME)/lib
+	$(RANLIB) $(DEST_Y_HOME)/lib/$(PKG_LIB)
 install-dll: $(PKG_DLL)
-	$(YNSTALL) $(PKG_DLL) $(Y_HOME)/lib
+	$(YNSTALL) $(PKG_DLL) $(DEST_Y_HOME)/lib
 
 uninstall::
-	if test $(PKG_EXE) != yorick; then rm -f $(Y_BINDIR)/$(PKG_EXE); fi
-	rm -f $(Y_HOME)/lib/$(PKG_LIB) $(Y_HOME)/lib/$(PKG_DLL)
-	cd $(Y_SITE)/i0; rm -f $(PKG_I)
-	if test -n "$(PKG_I_START)"; then cd $(Y_HOME)/i-start; rm -f $(PKG_I_START); fi
-	if test -n "$(PKG_I_EXTRA)"; then cd $(Y_SITE)/i; rm -f $(PKG_I_EXTRA); fi
+	if test $(PKG_EXE) != yorick; then rm -f $(DEST_Y_BINDIR)/$(PKG_EXE); fi
+	rm -f $(DEST_Y_HOME)/lib/$(PKG_LIB) $(DEST_Y_HOME)/lib/$(PKG_DLL)
+	cd $(DEST_Y_SITE)/i0; rm -f $(PKG_I)
+	if test -n "$(PKG_I_START)"; then cd $(DEST_Y_HOME)/i-start; rm -f $(PKG_I_START); fi
+	if test -n "$(PKG_I_EXTRA)"; then cd $(DEST_Y_SITE)/i; rm -f $(PKG_I_EXTRA); fi
 
 debug:
 	@$(MAKE) TGT=exe COPT=-g
